services/merak-topo/handler/handler.go

```diff
@@ -287,18 +286,17 @@ func UpdateComputenodeInfo(k8client *kubernetes.Clientset, topo_id string) error
 	// 	return fmt.Errorf("query mac error %s", err3)
 	// }
 
-	// log.Printf("updatecomputenode:=========Update mac addresses ===========")
-
 	for _, node := range topo.Vnodes {
 
 		if strings.Contains(node.Name, "vhost") {
 
 			var cnode database.ComputeNode
 
-			res, err := k8client.CoreV1().Pods("default").Get(Ctx, node.Name, metav1.GetOptions{})
+			res, err := k8client.CoreV1().Pods(namespace).Get(Ctx, node.Name, metav1.GetOptions{})
 
 			if err != nil {
-				log.Printf("updatecomputenode: get pod info from k8s error %s", err)
+				utils.Logger.Error("can't get pod info from k8s", node.Name, err.Error(), "namespace", namespace)
+				err_flag = 1
```

Comment:
@zzengcs same here...

Reply:
addressed-- return err when querying k8s cluster worker nodes does not work